qemu/tests/blockdev_snapshot_chains.py

```diff
@@ -26,17 +26,15 @@ def prepare_snapshot_file(self):
             params["image_size_%s" % snapshot_tag] = self.base_image.size
             params["image_name_%s" % snapshot_tag] = snapshot_tag
             self.params["image_name_%s" % snapshot_tag] = snapshot_tag
-            if self.params["image_backend"] == "iscsi_direct":
-                self.params.update({"enable_iscsi_%s" % snapshot_tag: "no"})
-                self.params.update({"image_raw_device_%s" % snapshot_tag: "no"})
             snapshot_format = params.get("snapshot_format", "qcow2")
             params["image_format_%s" % snapshot_tag] = snapshot_format
             self.params["image_format_%s" % snapshot_tag] = snapshot_format
-            if self.params["image_backend"] == "iscsi_direct":
+            backend = self.params["image_backend"]
+            if backend == "iscsi_direct":
                 self.params.update({"enable_iscsi_%s" % snapshot_tag: "no"})
                 self.params.update({"image_raw_device_%s" % snapshot_tag: "no"})
-            elif self.params["image_backend"] == "ceph":
-                self.params.update({"enable_ceph_%s" % snapshot_tag: "no"})
+            else:
+                self.params.update({"enable_%s_%s" % (backend, snapshot_tag): "no"})
```

Comment:
Your update can use short code to implement function with the ceph/nbd backend, but it's not that clear to understand the codes, and if we wants to support more backends or if we want to do more settings for a specific backend, we may also have to update the codes here. So what about coding here in the same way as we did for iscsi_direct?

Reply:
Have upated it, thanks zhenchao.